### PR TITLE
Integration tests for iOS RN App

### PR DESCRIPTION
**Pull request by ivanbogdanov**

### CODE REVIEW

File: ios/SmartSyncExplorerReactNativeTests/SmartSyncExplorerReactNativeTests.m [hunk 1]
```diff
@@ -0,0 +1,70 @@
+//
+//  FirstTest.m
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNativeTests/SmartSyncExplorerReactNativeTests.m]: Replace with standard `Mobile SDK` copyright.

File: ios/SmartSyncExplorerReactNative.xcodeproj/xcshareddata/xcschemes/SmartSyncExplorerReactNative.xcscheme [hunk 2]
```diff
@@ -62,11 +62,18 @@
...
+            value = "YES"
+            isEnabled = "YES">
+         </EnvironmentVariable>
+      </EnvironmentVariables>
       <AdditionalOptions>
       </AdditionalOptions>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/xcshareddata/xcschemes/SmartSyncExplorerReactNative.xcscheme]: Should be `Debug`. Revert this.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -411,6 +545,15 @@
...
 
 /* Begin XCConfigurationList section */
+		05B9FB241E03778D00712BD5 /* Build configuration list for PBXNativeTarget "SmartSyncExplorerReactNativeTests" */ = {
+			isa = XCConfigurationList;
+			buildConfigurations = (
+				05B9FB221E03778D00712BD5 /* Debug */,
+				05B9FB231E03778D00712BD5 /* Release */,
+			);
+			defaultConfigurationIsVisible = 0;
+			defaultConfigurationName = Release;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: Should be `Debug`.

File: IntegrationTests/testapp.ios.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (bhariharan) [inline IntegrationTests/testapp.ios.js]: Replace with standard `Mobile SDK` copyright.

File: IntegrationTests/IntegrationTestSmartStoreBridgeTest.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (bhariharan) [inline IntegrationTests/IntegrationTestSmartStoreBridgeTest.js]: Replace with standard `Mobile SDK` copyright.

File: IntegrationTests/testapp.ios.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+var TESTS = [
+  require('./IntegrationTestSmartStoreBridgeTest')
+];
+
+TESTS.forEach(
+  (test) => AppRegistry.registerComponent(test.displayName, () => test)
+);
+
+// Modules required for integration tests
+//require('LoggingTestModule');
```
Comment (bhariharan) [inline IntegrationTests/testapp.ios.js]: Remove commented out line.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -144,6 +196,12 @@
 				LastUpgradeCheck = 0810;
 				ORGANIZATIONNAME = SalesforceSamples;
 				TargetAttributes = {
+					05B9FB1A1E03778D00712BD5 = {
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj:199]: Remove this entry. There should be no `DevelopmentTeam` or `Provisioning...`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -257,7 +346,52 @@
...
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = D3C8BU4ASG;
+				INFOPLIST_FILE = SmartSyncExplorerReactNativeTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 10.1;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = com.salesforce.mobile.SmartSyncExplorerReactNativeTests;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: Rename to `com.salesforce.mobilesdk.SmartSyncExplorerReactNativeTests`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -257,7 +346,52 @@
...
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = D3C8BU4ASG;
+				INFOPLIST_FILE = SmartSyncExplorerReactNativeTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 10.1;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: This should be `9.0`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -257,7 +346,52 @@
...
 /* Begin XCBuildConfiguration section */
+		05B9FB221E03778D00712BD5 /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = D3C8BU4ASG;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: Remove `DEVELOPMENT_TEAM` entry.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -257,7 +346,52 @@
...
+		05B9FB231E03778D00712BD5 /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				COPY_PHASE_STRIP = NO;
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				DEVELOPMENT_TEAM = D3C8BU4ASG;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: Remove `DEVELOPMENT_TEAM` entry.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -257,7 +346,52 @@
...
+				INFOPLIST_FILE = SmartSyncExplorerReactNativeTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 10.1;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: Should be `9.0`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -257,7 +346,52 @@
...
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = com.salesforce.mobile.SmartSyncExplorerReactNativeTests;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: `com.salesforce.mobilesdk.SmartSyncExplorerReactNativeTests`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/xcshareddata/xcschemes/SmartSyncExplorerReactNative.xcscheme [hunk 14]
```diff
@@ -40,13 +40,13 @@
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      shouldUseLaunchSchemeArgsEnv = "YES">
+      shouldUseLaunchSchemeArgsEnv = "NO">
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/xcshareddata/xcschemes/SmartSyncExplorerReactNative.xcscheme]: Should be `YES`?

File: ios/SmartSyncExplorerReactNative/AppDelegate.m [hunk 15]
```diff
@@ -138,6 +139,27 @@ - (void)initializeAppViewState
...
+  }
+  else if (!getenv("CI_USE_PACKAGER")) {
+#ifdef DEBUG
+    [self log:SFLogLevelDebug msg:@"Is debug mode."];
+#else
+    jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+#endif
+
+  }
+
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative/AppDelegate.m]: Remove extra empty lines here.

**Review by bhariharan — CHANGES_REQUESTED** [review]
@ivanbogdanov Just a few changes.

**Comment by bhariharan** [comment]
@ivanbogdanov Added some comments.

**Comment by ivanbogdanov** [comment]
@bhariharan working on fixes. Planning to update the PR soon

**Comment by ivanbogdanov** [comment]
@bhariharan I think I am done with fixes. Thank you for looking into it.

File: IntegrationTests/IntegrationTestSmartStoreBridgeTest.js [hunk 16]
```diff
@@ -0,0 +1,99 @@
+/*
+ Copyright (c) 2011-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline IntegrationTests/IntegrationTestSmartStoreBridgeTest.js]: `2016-present` :-)

File: IntegrationTests/testapp.ios.js [hunk 17]
```diff
@@ -0,0 +1,83 @@
+/*
+ Copyright (c) 2011-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline IntegrationTests/testapp.ios.js]: `2016-present` :-)

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 18]
```diff
@@ -257,7 +345,52 @@
...
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = "";
+				INFOPLIST_FILE = SmartSyncExplorerReactNativeTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = com.salesforce.mobile.SmartSyncExplorerReactNativeTests;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: `com.salesforce.mobilesdk.SmartSyncExplorerReactNativeTests`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 19]
```diff
@@ -257,7 +345,52 @@
...
 /* Begin XCBuildConfiguration section */
+		05B9FB221E03778D00712BD5 /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = "";
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: You might have to manually remove the `DEVELOPMENT_TEAM` line with a text editor.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 20]
```diff
@@ -257,7 +345,52 @@
...
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				COPY_PHASE_STRIP = NO;
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				DEVELOPMENT_TEAM = "";
+				INFOPLIST_FILE = SmartSyncExplorerReactNativeTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = com.salesforce.mobile.SmartSyncExplorerReactNativeTests;
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: `com.salesforce.mobilesdk.SmartSyncExplorerReactNativeTests`.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 21]
```diff
@@ -257,7 +345,52 @@
...
+		05B9FB231E03778D00712BD5 /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_SUSPICIOUS_MOVES = YES;
+				COPY_PHASE_STRIP = NO;
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				DEVELOPMENT_TEAM = "";
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: You might have to manually remove the `DEVELOPMENT_TEAM` line with a text editor.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 22]
```diff
@@ -267,6 +400,7 @@
 				CODE_SIGN_ENTITLEMENTS = SmartSyncExplorerReactNative/SmartSyncExplorerReactNative.entitlements;
 				CURRENT_PROJECT_VERSION = 1;
 				DEAD_CODE_STRIPPING = NO;
+				DEVELOPMENT_TEAM = "";
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: You might have to manually remove the `DEVELOPMENT_TEAM` line with a text editor.

File: ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj [hunk 23]
```diff
@@ -295,6 +429,7 @@
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
 				CODE_SIGN_ENTITLEMENTS = SmartSyncExplorerReactNative/SmartSyncExplorerReactNative.entitlements;
 				CURRENT_PROJECT_VERSION = 1;
+				DEVELOPMENT_TEAM = "";
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNative.xcodeproj/project.pbxproj]: You might have to manually remove the `DEVELOPMENT_TEAM` line with a text editor.

File: ios/SmartSyncExplorerReactNativeTests/SmartSyncExplorerReactNativeTests.m [hunk 24]
```diff
@@ -0,0 +1,86 @@
+/*
+ Copyright (c) 2011-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ios/SmartSyncExplorerReactNativeTests/SmartSyncExplorerReactNativeTests.m]: `2016-present`.

**Review by bhariharan — CHANGES_REQUESTED** [review]
@ivanbogdanov Just a few more minor changes.

File: IntegrationTests/IntegrationTestSmartStoreBridgeTest.js (hunk 16 above)
Comment (ivanbogdanov) [reply IntegrationTests/IntegrationTestSmartStoreBridgeTest.js]: Interesting: I have copied it from AppDelegate :)

Comment (bhariharan) [reply IntegrationTests/IntegrationTestSmartStoreBridgeTest.js]: Yeah, that was created in `2011` :-)

**Comment by ivanbogdanov** [comment]
Done. Let me know.

**Review by bhariharan — APPROVED** [review]
👍

**Pull request merged**